### PR TITLE
Added Jump Search Algorithm

### PR DESCRIPTION
**Pull request by Kaixm**
Hi, I have added the Jump Search algorithm and tests. Please have a look, and do let me know if there are any issues.

- [x] I have performed a self-review of my code
- [x] My code follows the style guidelines of this project
- [x] I have added tests that prove my fix is effective or that my feature works
- [x] New and existing unit tests pass locally with my changes
- [x] Comments in areas I changed are up to date
- [x] I have added comments to hard-to-understand areas of my code
- [x] I have made corresponding changes to the README.md
- [x] I have made corresponding changes to the DIRECTORY.md

### CODE REVIEW

File: Algorithms/Search/JumpSearcher.cs [hunk 1]
```diff
@@ -0,0 +1,54 @@
+using System;
+
+namespace Algorithms.Search
+{
+    /// <summary>
+    ///     Jump Search checks fewer elements by jumping ahead by fixed steps. 
+    ///     The optimal steps to jump is √n, where n refers to the number of elements in the array.
+    ///     Time Complexity: O(√n)
```
Comment (siriak) [inline Algorithms/Search/JumpSearcher.cs:8]: Ha, looks interesting :)

File: Algorithms/Search/JumpSearcher.cs [hunk 2]
```diff
@@ -0,0 +1,54 @@
...
+        ///     Find the index of the item searched for in the array.
+        /// </summary>
+        /// <param name="sortedArray">Sorted array to be search in. Cannot be empty or null.</param>
+        /// <param name="searchItem">Item to be search for. Cannot be null.</param>
+        /// <returns>If item is found, return index. Else, return -1.</returns>
+        public int FindIndex(T[] sortedArray, T searchItem)
+        {
+            if (sortedArray.Length == 0 || sortedArray is null || searchItem is null)
+            {
+                throw new NullReferenceException();
```
Comment (siriak) [inline Algorithms/Search/JumpSearcher.cs:24]: Please use ArgumentNullException for null cases and return -1 for empty array

**Review by siriak — COMMENTED** [review]
Thanks for contributing! Please fix issues Codacy found and other automatic checks

**Comment by Kaixm** [comment]
I have fixed the mentioned issues and passed all the checks, do let me know if there are any other changes needed!

**Review by siriak — APPROVED** [review]
LGTM, thanks!

**Pull request merged**